File: app/src/main/java/uk/co/ribot/androidboilerplate/injection/module/FragmentModule.java
package uk.co.ribot.androidboilerplate.injection.module;

import android.content.Context;
import android.support.v4.app.Fragment;

import com.sunny.common.injection.qualifier.FragmentContext;

import dagger.Module;
import dagger.Provides;

@Module
public class FragmentModule {

    private Fragment mFragment;

    public FragmentModule(Fragment fragment) {
        mFragment = fragment;
    }

    @Provides
    Fragment provideFragment() {
        return mFragment;
    }

    @Provides
    @FragmentContext
    Context providesContext() {
        return mFragment.getContext();
    }
}

File: app/src/main/java/uk/co/ribot/androidboilerplate/injection/module/FragmentModule.kt
package uk.co.ribot.androidboilerplate.injection.module

import android.content.Context
import android.support.v4.app.Fragment

import dagger.Module
import dagger.Provides
import uk.co.ribot.androidboilerplate.injection.qualifier.FragmentContext

@Module
class FragmentModule(private val mFragment: Fragment) {

    @Provides
    internal fun provideFragment(): Fragment {
        return mFragment
    }

    @Provides
    @FragmentContext
    internal fun providesContext(): Context {
        return mFragment.context!!
    }
}
